src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -827,3 +1069,108 @@ def __removed_older_complete_status_files(self, status_folder):
 
         self.composite_logger.log_debug("Cleaned up older complete status files: {0}".format(files_removed))
 
+    def __calc_status_size_on_disk(self, status_file_dumps):
+        """ Calculate status file size in bytes on disk """
+        return len(status_file_dumps.encode("utf-8"))
+
+    def __calc_package_payload_size_on_disk(self, package_list):
+        """ Calculate final package list size in bytes (because of escape chars) """
+        first_json_dump = json.dumps(package_list)
+        return len(json.dumps(first_json_dump).encode("utf-8"))
+
+    def size_of_constant_status_data(self, complete_status_file_payload, assessment_status_index, installation_status_index):
+        """ Get the size in bytes of the complete_status_file without packages data  """
+        status_file_no_list_data = complete_status_file_payload
+        if assessment_status_index is not None:
+            assessment_msg_without_packages = self.__update_substatus_msg(substatus_msg=self.__assessment_substatus_msg_copy, substatus_msg_patches=[])
+            status_file_no_list_data['status']['substatus'][assessment_status_index]['formattedMessage']['message'] = json.dumps(assessment_msg_without_packages)
+
+        if installation_status_index is not None:
+            installation_msg_without_packages = self.__update_substatus_msg(substatus_msg=self.__installation_substatus_msg_copy, substatus_msg_patches=[])
+            status_file_no_list_data['status']['substatus'][installation_status_index]['formattedMessage']['message'] = json.dumps(installation_msg_without_packages)
+
+        return self.__calc_status_size_on_disk(json.dumps(status_file_no_list_data))
+
+    def __get_substatus_index(self, substatus_name, substatus):
+        """" Get substatus index from the current substatus """
+        for index, sub_name in enumerate(substatus):
```

Comment:
sub_name indicates it contains only the substatus name, whereas in here, sub_name contains the entire substatus. So the naming here is not correct. Please rename 'sub_name' to substatus and substatus can be renamed to substatus_list

Reply:
done